sale_order_type/models/account_move.py

```diff
@@ -17,7 +19,7 @@ class AccountMove(models.Model):
         copy=True,
     )
 
-    @api.depends("partner_id", "company_id")
+    @api.depends("partner_id")
```

Comment:
Although I understand the reason to remove that on the use case, this is limiting the correct logic. Why the company is always being assigned no matter if it's the same? Isn't that what we have to tackle?